esx_service/vmdk_ops.py

```diff
@@ -402,8 +402,15 @@ def vol_info(vol_meta, vol_size_info, datastore):
 # Return error, or None for OK
 def removeVMDK(vmdk_path):
     logging.info("*** removeVMDK: %s", vmdk_path)
-    cmd = "{0} {1}".format(VMDK_DELETE_CMD, vmdk_path)
 
+    # Check the current volume status
+    attached, uuid, attach_as = getStatusAttached(vmdk_path)
+    if attached:
```

Comment:
@pdhamdhere
Yes, the real problem is vmkfstools unlinking associated vmdk files without any verification. I still didn't had time to test with the vim.virtualdiskmanager and wasn't aware of the -D flag. Will try to do this later today.

Maybe we also need an background async mechanism to detect and update the cases of stale metadata and avoid being compelled to no trust it?